rails/app/controllers/api/v1/classes_controller.rb

```diff
@@ -2,32 +2,24 @@ class API::V1::ClassesController < API::APIController
 
   # GET api/v1/classes/:id
   def show
-    auth = auth_student_or_teacher_or_researcher(params)
```

Comment:
I think this will break the access grant/jwt access to this controller.  The `auth_student_or_teacher_or_researcher` method called the `auth_not_anonymous` method which called `check_for_auth_token` to check for a bearer token.